dmutils/flask_init.py

```diff
@@ -1,7 +1,33 @@
+from collections import OrderedDict
+from itertools import chain
 import os
-from . import config, logging, proxy_fix, request_id, formats, filters, errors
+from types import MappingProxyType
+
+from dmutils import config, logging, proxy_fix, request_id, formats, filters
+from dmutils.errors import api as api_errors, frontend as fe_errors
 from flask_script import Manager, Server
 from flask_wtf.csrf import CSRFError
+from werkzeug.exceptions import default_exceptions
+
+
+frontend_error_handlers = MappingProxyType(OrderedDict((
+    (CSRFError, fe_errors.csrf_handler,),
+    (400, fe_errors.render_error_page,),
+    (401, fe_errors.redirect_to_login,),
+    (403, fe_errors.redirect_to_login,),
+    (404, fe_errors.render_error_page,),
+    (410, fe_errors.render_error_page,),
+    (503, fe_errors.render_error_page,),
+    (500, fe_errors.render_error_page,),
+)))
+
+
+api_error_handlers = MappingProxyType(OrderedDict(chain(
```

Comment:
What are we using the `MappingProxyType` for here? Can we just use a dict (not even an ordered one I think)? Also you can just `+` tuples together rather than importing `chain`.


```
api_error_handlers = MappingProxyType(OrderedDict(chain(
    (
        (api_errors.ValidationError, api_errors.validation_error_handler,),
    ),
    ((code, api_errors.json_error_handler) for code in default_exceptions),
)))
```

```
dict(((api_errors.ValidationError, api_errors.validation_error_handler),) + ((code, api_errors.json_error_handler) for code in default_exceptions)
```

Reply:
Uh yeah we could probably do `((api_errors.ValidationError, api_errors.validation_error_handler,),) + tuple((code, api_errors.json_error_handler) for code in default_exceptions)`

As for `MappingProxyType`, it just makes the dict immutable, which makes it easier to reason about and harder to do weird and unexpected things to.